var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -126,6 +126,11 @@ class Hdf5(CMakePackage):
         sha256="81f6201aba5c30dced5dcd62f5d5477a2790fd5850e02ac514ca8bf3e2bb375a",
         preferred=True,
     )
+    version(
+        "1.8.23",
+        sha256="37fa4eb6cd0e181eb49a10d54611cb00700e9537f805d03e6853503afe5abc27",
+        preferred=True,
+    )
```

Comment:
Is there any reason why _all_ release versions of `hdf5` are preferred?

Reply:
Confirmed checksum

Reply:
> Is there any reason why _all_ release versions of `hdf5` are preferred?

They are all _that good_. I don't know. In making my patch, I just followed the pattern.

Reply:
This was done when the 1.13.1 development version was added so that a package conflicting with HDF5 1.14.x would default to the previous stable release.  See the request for adding preferred=True to all "maintenance versions" and discussion in PR 29514.  The 1.13.x versions have been removed, so it might no longer be needed.